server/server.go

```diff
@@ -943,3 +948,79 @@ func isNotFound(err error) bool {
 
 	return false
 }
+
+// startReloadWatcherForMirrorRegistries starts a new goroutine that
+// monitors the "registries.conf.d" directory. If not set, it defaults to
+// "/etc/containers/registries.conf.d". The routine reloads the registries
+// configuration when a file is modified, created or deleted in the directory.
+func (s *Server) startReloadWatcherForMirrorRegistries(ctx context.Context, registriesConfDDir string) {
+	if registriesConfDDir == "" {
+		log.Infof(ctx, "No registries.conf.d directory specified, defaulting to /etc/containers/registries.conf.d")
+		registriesConfDDir = "/etc/containers/registries.conf.d"
+	}
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Fatalf(ctx, "Failed to create new watcher: %v", err)
+		return
+	}
+	defer watcher.Close()
+
+	log.Infof(ctx, "Registered reload watcher for mirror registries configuration")
+	done := make(chan struct{})
+	go s.reloadMirrorRegistriesConfiguration(ctx, watcher, done, registriesConfDDir)
```

Comment:
Do we need this additional goroutine? We just create the routine and then wait for `done`. Everything could be inline from my point of view since we're already in a new go routine.